monitors/test/ContractMonitor.js

```diff
@@ -0,0 +1,295 @@
+
+// Truffle artifacts
+const ExpiringMultiParty = artifacts.require("ExpiringMultiParty");
+const Finder = artifacts.require("Finder");
+const IdentifierWhitelist = artifacts.require("IdentifierWhitelist");
+const MockOracle = artifacts.require("MockOracle");
+const TokenFactory = artifacts.require("TokenFactory");
+const Token = artifacts.require("ExpandedERC20");
+const Timer = artifacts.require("Timer");
+
+contract("ContractMonitor.js", function(accounts) {
+  const tokenSponsor = accounts[0];
+  const liquidator = accounts[1];
+  const disputer = accounts[2];
+  const sponsor1 = accounts[3];
+  const sponsor2 = accounts[4];
+
+  const zeroAddress = "0x0000000000000000000000000000000000000000";
+
+  // Contracts
+  let collateralToken;
+  let collateralTokenSymbol;
+  let emp;
+  let syntheticToken;
+  let mockOracle;
+  let identifierWhitelist;
+
+  // Test object for EMP event client
+  let eventClient;
+
+  // re-used variables
+  let expirationTime;
+  let constructorParams;
+
+  const spy = sinon.spy();
+
+  before(async function() {
+    collateralToken = await Token.new("Dai Stable coin", "Dai", 18, { from: tokenSponsor });
+
+    identifierWhitelist = await IdentifierWhitelist.deployed();
+    await identifierWhitelist.addSupportedIdentifier(web3.utils.utf8ToHex("UMATEST"));
+
+    // Create a mockOracle and finder. Register the mockOracle with the finder.
+    finder = await Finder.deployed();
+    mockOracle = await MockOracle.new(finder.address, Timer.address);
+    const mockOracleInterfaceName = web3.utils.utf8ToHex(interfaceName.Oracle);
+    await finder.changeImplementationAddress(mockOracleInterfaceName, mockOracle.address);
+  });
+
+  beforeEach(async function() {
+    const currentTime = await mockOracle.getCurrentTime.call();
+    const timer = await Timer.deployed();
+    await timer.setCurrentTime(currentTime.toString());
+    expirationTime = currentTime.toNumber() + 100; // 100 seconds in the future
+
+    constructorParams = {
+      isTest: true,
+      expirationTimestamp: expirationTime.toString(),
+      withdrawalLiveness: "1000",
+      collateralAddress: collateralToken.address,
+      finderAddress: Finder.address,
+      tokenFactoryAddress: TokenFactory.address,
+      timerAddress: Timer.address,
+      priceFeedIdentifier: web3.utils.utf8ToHex("UMATEST"),
+      syntheticName: "Test UMA Token",
+      syntheticSymbol: "UMATEST",
+      liquidationLiveness: "10",
+      collateralRequirement: { rawValue: toWei("1.5") },
+      disputeBondPct: { rawValue: toWei("0.1") },
+      sponsorDisputeRewardPct: { rawValue: toWei("0.1") },
+      disputerDisputeRewardPct: { rawValue: toWei("0.1") },
+      minSponsorTokens: { rawValue: toWei("1") }
+    };
+
+    // Create a sinon spy and give it to the SpyTransport as the winston logger. Use this to check all winston
+    // logs the correct text based on interactions with the emp.
+
+    const spyLogger = winston.createLogger({
+      level: "info",
```

Comment:
Does this mean the `spy` will only test the "info" level logs?

Reply:
If so, can you add a clarifying comment about how the SpyTransport should be constructed in order to test as intended?

Reply:
yup. it only tests `info` level. I'll add a comment